lib/ocpp/v201/device_model_storage_sqlite.cpp

```diff
@@ -20,6 +20,12 @@ DeviceModelStorageSqlite::DeviceModelStorageSqlite(const fs::path& db_path) {
     }
 }
 
+DeviceModelStorageSqlite::~DeviceModelStorageSqlite() {
+    if (sqlite3_close(this->db) != SQLITE_OK) {
```

Comment:
I would take a look at how its done in `close_connection_internal` in the `database_connection.cpp` file. You could also finalize all open transactions and use `_v2` close.

Reply:
I think the best way forward would be to refactor this a bit and use the database_connection which already takes care of many of the details that you have mentioned.

Reply:
Yes I think you are right, just didn't want to blow up the PR like that. In the long run this is better :+1: